store/src/java/com/zimbra/cs/util/proxyconfgen/ProxyConfGen.java

```diff
@@ -2401,6 +2414,32 @@ public static int createConf(String[] args) throws ServiceException, ProxyConfEx
     return (exitCode);
   }
 
+  /**
+   * Cleanup of Obsolete certificate-key file pair from {@link ProxyConfGen#DOMAIN_SSL_DIR}
+   *
+   * @param mDomainReverseProxyAttrs List<{@link DomainAttrItem}> domain attribute items collected
+   *     from domains
+   * @param dryRun if it's a dry run
+   * @author Keshav Bhatt
+   * @since 22.12.0
+   */
+  private static void deleteObsoleteCertificateKeyPair(
+      final List<DomainAttrItem> mDomainReverseProxyAttrs, final boolean dryRun) {
+    List<String> filesInDirectory = Utils.getFilesPathInDirectory(DOMAIN_SSL_DIR);
+    for (DomainAttrItem entry : mDomainReverseProxyAttrs) {
+      filesInDirectory.removeIf(filePath -> (filePath.contains(entry.domainName)));
```

Comment:
Sorry but doesn't this delete a certificate that currently exists? Souldn't it be
```suggestion
      filesInDirectory.removeIf(filePath -> !(filePath.contains(entry.domainName)));
```
? Note that the certificate is moved from `certificate.crt` to `certificate.crt.bak` before every run, overwriting the old `bak` one.

Reply:
Oh yeah right, I'm dumb :sweat_smile: